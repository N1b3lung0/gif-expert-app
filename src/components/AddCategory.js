import React, { useState } from 'react'
import PropTypes from 'prop-types';
import { InputText } from 'primereact/inputtext';


export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => setInputValue( e.target.value );

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories( categories => [inputValue, ...categories] );
            setInputValue('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <span className="p-float-label">
                <InputText
                    id="float-input"
                    type="text"
                    value={ inputValue }
                    onChange={ handleInputChange }
                />
                <label htmlFor="float-input">Search</label>
            </span>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}