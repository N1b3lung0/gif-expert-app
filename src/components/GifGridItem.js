import 'primereact/resources/themes/nova-dark/theme.css';
import 'primereact/resources/primereact.css';

import React from 'react'
import PropTypes from 'prop-types';
import { Card } from 'primereact/card';

export const GifGridItem = ({ title, url}) => {
    
    const header = ( <img src={ url } alt={ title }/> );

    return (
        <Card
            subTitle={ title }
            style={{width: '360px'}}
            className="ui-card-shadow card animate__animated animate__fadeIn"
            header={ header }
        />
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}

Card.propTypes = {
    subTitle: PropTypes.string.isRequired,
    header: PropTypes.object.isRequired
}
