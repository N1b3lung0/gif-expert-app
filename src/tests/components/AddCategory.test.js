import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';
import { InputText } from 'primereact/inputtext';

describe('Testing AddCategory component', () => {

    const setCategories = jest.fn();
    const id = 'id';
    const type = 'text';
    let value = '';
    const onChange = jest.fn();
    let wrapperAddCategory = shallow( <AddCategory setCategories={ setCategories } /> );
    let wrapperInputText = shallow( <InputText id={ id } type={ type } value={ value } onChange={ onChange } /> );

    beforeEach( () => {
        jest.clearAllMocks();
        wrapperAddCategory = shallow( <AddCategory setCategories={ setCategories } /> );
        wrapperInputText = shallow( <InputText id={ id } type={ type } value={ value } onChange={ onChange } /> );
    });
    
    test('should show the AddCategory component correctly', () => {
        expect(wrapperAddCategory).toMatchSnapshot();
    });

    test('should show the InputText component correctly', () => {
        expect(wrapperInputText).toMatchSnapshot();
        wrapperInputText.simulate('change', { target: { value } });
        expect(wrapperInputText.prop('id')).toBe(id);
        expect(wrapperInputText.prop('type')).toBe(type);
        expect(wrapperInputText.prop('value')).toBe(value);
        expect(wrapperInputText.prop('onChange')).toBe(onChange);
    });

    test('should NOT post the info on submit', () => {
        wrapperAddCategory.find('form').simulate('submit', { preventDefault(){} });
        expect(setCategories).not.toHaveBeenCalled();
    });
    
    test('should call to setCategories and clean the inputText', () => {
        value = 'Hola Mundo'
        wrapperAddCategory.find('InputText').simulate('change', { target: { value } });
        wrapperAddCategory.find('form').simulate('submit', { preventDefault(){} });
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect(wrapperAddCategory.find('InputText').prop('value')).toBe('');
    });
    
    
})
