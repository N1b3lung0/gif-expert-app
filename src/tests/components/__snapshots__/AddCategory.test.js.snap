// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddCategory component should show the AddCategory component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <span
    className="p-float-label"
  >
    <InputText
      id="float-input"
      keyfilter={null}
      onChange={[Function]}
      onInput={null}
      onKeyPress={null}
      tooltip={null}
      tooltipOptions={null}
      type="text"
      validateOnly={false}
      value=""
    />
    <label
      htmlFor="float-input"
    >
      Search
    </label>
  </span>
</form>
`;

exports[`Testing AddCategory component should show the InputText component correctly 1`] = `
<input
  className="p-inputtext p-component"
  id="id"
  onChange={[MockFunction]}
  onInput={[Function]}
  onKeyPress={[Function]}
  type="text"
  value=""
/>
`;
